JASP-Engine/JASP/R/auditClassicalBenfordsLaw.R

```diff
@@ -0,0 +1,579 @@
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+# When making changes to this file always mention @koenderks as a 
+# reviewer in the Pull Request
+
+auditClassicalBenfordsLaw <- function(jaspResults, dataset, options, ...){
+
+  # Create the procedure paragraph
+  .auditBenfordsLawProcedureParagraph(options, 
+                                      jaspResults,
+                                      position = 1)
+  
+  # Read in the data 
+  dataset <- .auditReadDataBenfordsLaw(dataset, 
+                                       options)
+
+  # Perform early error checks
+  .auditClassicalBenfordsLawErrorCheck(dataset, 
+                                       options)
+
+  # Ready for analysis
+  ready <- .auditClassicalBenfordsLawReady(options)
+
+  benfordsLawContainer <- .auditBenfordsLawGetContainer(jaspResults, 
+                                                        position = 2)
+
+  # --- TABLES
+
+  # Create a state to keep track of table numbers
+  .auditCreateTableNumber(jaspResults)
+  
+  # Create the goodness-of-fit table
+  .auditClassicalBenfordsLawTestTable(dataset, 
+                                    options, 
+                                    benfordsLawContainer, 
+                                    jaspResults,
+                                    ready, 
+                                    positionInContainer = 1)
+
+  # Create the observed and predicted probabilities table                                  
+  .auditClassicalBenfordsLawSummaryTable(dataset, 
+                                         options, 
+                                         benfordsLawContainer, 
+                                         jaspResults,
+                                         ready, 
+                                         positionInContainer = 2)
+
+  # ---
+
+  # --- PLOTS
+
+  .auditCreateFigureNumber(jaspResults)
+  
+  # Create the observed and predicted probabilities plot
+  .benfordsLawPlot(dataset, 
+                   options, 
+                   benfordsLawContainer, 
+                   jaspResults,
+                   ready, 
+                   positionInContainer = 3)
+
+  # ---
+
+  # Create the conclusion paragraph
+  .auditBenfordsLawConclusionParagraph(options,
+                                       benfordsLawContainer,
+                                       jaspResults,
+                                       ready,
+                                       position = 3)
+  
+  # --- BADGES
+
+  # Provide the analysis badges
+  .auditBadgeSection(options,
+                     type = "benfordsLaw",
+                     stateContainer = benfordsLawContainer,
+                     jaspResults, 
+                     ready, 
+                     position = 4)
+
+  # ---
+}
+
+.auditReadDataBenfordsLaw <- function(dataset, 
+                                     options){
+
```

Comment:
Would be good to already prep this for the dataset being delivered by JASP (which will happen in the future)
```suggestion
if (!is.null(dataset)) return(dataset)
```